internal/api/api.go

```diff
@@ -72,64 +74,90 @@ func OrgsApiURL(baseURL, slugName string) (*url.URL, error) {
-	res, err := a.client.Get(u)
+	selfData, err := a.GetSelf()
 	if err != nil {
 		return "", fmt.Errorf("unable to retrieve org ID: %w", err)
 	}
-	//goland:noinspection GoUnhandledErrorResult
-	defer res.Body.Close()
 
-	body, err := io.ReadAll(res.Body)
-	if err != nil {
-		return "", fmt.Errorf("unable to retrieve org ID: %w", err)
-	}
+	return selfData.Data.Attributes.DefaultOrgContext, nil
+}
 
-	if res.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("unable to retrieve org ID (status: %d)", res.StatusCode)
+func (a *snykApiClient) GetUserMe() (string, error) {
+	selfData, err := a.GetSelf()
+	if err != nil {
+		return "", fmt.Errorf("error while fetching self data: %w", err) // Prioritize error
 	}
 
-	var userInfo contract.SelfResponse
-	if err = json.Unmarshal(body, &userInfo); err != nil {
-		return "", fmt.Errorf("unable to retrieve org ID (status: %d): %w", res.StatusCode, err)
+	if len(selfData.Data.Attributes.Username) == 0 {
+		return "", fmt.Errorf("error while extracting user: missing property 'username'")
 	}
 
-	return userInfo.Data.Attributes.DefaultOrgContext, nil
+	return selfData.Data.Attributes.Username, nil
 }
 
 func (a *snykApiClient) GetFeatureFlag(flagname string, orgId string) (bool, error) {
 	const defaultResult = false
 
-	u := a.url + "/v1/cli-config/feature-flags/" + flagname + "?org=" + orgId
+	endpoint := "/v1/cli-config/feature-flags/" + flagname + "?org=" + orgId
 
 	if len(orgId) <= 0 {
 		return defaultResult, fmt.Errorf("failed to lookup feature flag with orgiId not set")
 	}
 
-	res, err := a.client.Get(u)
-	if err != nil {
-		return defaultResult, fmt.Errorf("unable to retrieve feature flag: %w", err)
-	}
-	//goland:noinspection GoUnhandledErrorResult
-	defer res.Body.Close()
-
-	body, err := io.ReadAll(res.Body)
+	body, err := clientGet(a, endpoint)
 	if err != nil {
 		return defaultResult, fmt.Errorf("unable to retrieve feature flag: %w", err)
 	}
 
 	var flag contract.OrgFeatureFlagResponse
 	if err = json.Unmarshal(body, &flag); err != nil {
-		return defaultResult, fmt.Errorf("unable to retrieve feature flag (status: %d): %w", res.StatusCode, err)
+		return defaultResult, fmt.Errorf("unable to retrieve feature flag: %w", err)
 	}
 
-	if res.StatusCode != http.StatusOK || flag.Code == http.StatusUnauthorized || flag.Code == http.StatusForbidden {
+	if flag.Code == http.StatusUnauthorized || flag.Code == http.StatusForbidden {
 		return defaultResult, err
 	}
 
 	return true, nil
 }
 
+func (a *snykApiClient) GetSelf() (contract.SelfResponse, error) {
+	endpoint := constants.SNYK_API_SELF + constants.SNYK_API_VERSION
+	var selfData contract.SelfResponse
+
+	body, err := clientGet(a, endpoint)
+	if err != nil {
+		return selfData, err
+	}
+
+	if err = json.Unmarshal(body, &selfData); err != nil {
+		return selfData, fmt.Errorf("unable to retrieve self data: %w", err)
+	}
+
+	return selfData, nil
+}
+
+func clientGet(a *snykApiClient, endpoint string) ([]byte, error) {
```

Comment:
**suggestion:** Extend interface to include version as attribute and move URL building into clientGet. eg endpoint, version => `${endpoint}?version=${version}`